src/hooks/__tests__/use-file-upload-reducer.test.ts

```diff
@@ -0,0 +1,127 @@
+import {renderHook, act} from '@testing-library/react-hooks'
+import useFileUploadReducer from '../use-file-upload-reducer'
+
+test('it initializes with no files]', () => {
+  const {result} = renderHook(() => useFileUploadReducer([]))
+
+  const [state, _dispatch] = result.current
+
+  expect(state).toEqual({files: []})
+})
+
+test('it can add a file upload', () => {
+  const {result} = renderHook(() => useFileUploadReducer([]))
+
+  const initialState = result.current[0]
+  expect(initialState).toEqual({files: []})
+
+  const file: File = {name: 'fake-file.txt'} as File
+
+  const fileUpload = {
+    file,
+    percent: 0,
+    message: 'waiting to upload',
+  }
+
+  act(() => {
+    const dispatch = result.current[1]
+    dispatch({type: 'add', fileUpload})
+  })
+
+  const stateWithFile = result.current[0]
+
+  expect(stateWithFile).toEqual({files: [fileUpload]})
+})
+
+test('it can update progress of a file upload', () => {
+  const {result} = renderHook(() => useFileUploadReducer([]))
+
+  const initialState = result.current[0]
+  expect(initialState).toEqual({files: []})
```

Comment:
suggestions(testing): i'd remove these in all the tests that aren't "initial state is an empty files array" 

then failures are related to the stated intention of the test. the rule I use is 1 expect per test, with very few exceptions 

which is related to the setup of all these tests which appears to be similar so might be moved into a setup function